### PR TITLE
test(api): added v3 product catalog e2e tests

### PR DESCRIPTION
**Pull request by rolosp**
Overview

  - api/v3/handlers/plans/convert.go: removed early currency validation in
  FromAPICreatePlanRequest; the service layer's PlanMeta.Validate() already produces
   a structured ValidationIssue with the correct HTTP status and validationErrors
  extensions — behaviour from the client's perspective is unchanged (still 400), but
   the error shape is now consistent with all other validation errors
  - openmeter/productcatalog/planaddon/service/planaddon.go: UpdatePlanAddon now
  validates the post-merge state before persisting — builds a PlanAddon reflecting
  the patched fields (applying nil = keep-existing semantics for
  FromPlanPhase/Metadata/Annotations, nil = clear for MaxQuantity) and calls
  PlanAddon.Validate(), returning a structured GenericValidationError on failure

  e2e tests (e2e/)

  - v3helpers_test.go: shared v3Client struct, typed request/response helpers,
  fixture builders (validAddonRequest, validFlatRateCard, validUnitRateCard,
  validGraduatedRateCard, etc.), and a withPageSize option
  - addons_v3_test.go: full addon lifecycle (create → get → list → update → publish
  → archive → delete → soft-delete) plus mixed rate-card round-trip and
  versioning/auto-archive tests
  - plans_v3_test.go: plan lifecycle, versioning, and auto-archive tests
  - planaddons_v3_test.go: plan-addon attachment, status-matrix validation, and
  lifecycle tests

  Skill

  Added .agents/skills/e2e/SKILL.md documenting the v3 e2e test patterns for future
  agents.

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive E2E testing guide with v1/v3 patterns and best practices.
  * Clarified Plan-Addon MaxQuantity semantics (nil = unlimited; positive integer required when provided).
  * Updated generated API documentation payload.

* **Tests**
  * Added extensive V3 E2E suites covering plans, addons, plan-addon lifecycles, validation matrices, versioning, and edge cases.
  * Introduced V3 HTTP client and test helpers for stable live-server tests.

* **Bug Fixes**
  * Enforced plan-addon compatibility validation during updates.
  * Strengthened currency validation coverage for plan creation.

### CODE REVIEW

File: api/spec/packages/aip/src/productcatalog/planaddon.tsp [hunk 1]
```diff
@@ -27,7 +27,9 @@ model PlanAddon {
 
   /**
    * The maximum number of times the add-on can be purchased for the plan.
-   * Not applicable for single instance add-ons.
+   * Omitting this field (null) means unlimited — no cap is enforced.
+   * Must be a positive integer when set; zero and negative values are rejected.
+   * Not applicable for single-instance add-ons; must be omitted for those.
```
Comment (tothandras) [inline api/spec/packages/aip/src/productcatalog/planaddon.tsp]:
```suggestion
   * For single-instance add-ons this field must be omitted.
   * For multi-instance add-ons when omitted, unlimited quantity can be purchased.
```

File: api/v3/handlers/plans/convert.go [hunk 2]
```diff
@@ -407,12 +407,12 @@ func FromAPICreatePlanRequest(ns string, body api.CreatePlanRequest) (plan.Creat
...
-	if err := cur.Validate(); err != nil {
-		return req, fmt.Errorf("invalid currency: %w", err)
-	}
-
-	req.Currency = cur
+	// Currency is deliberately not validated here — the service layer's
+	// PlanMeta.Validate() produces a structured ValidationIssue
+	// (ErrCurrencyInvalid, critical severity + 400 HTTP attribute) so an
+	// unknown currency surfaces as 400 with extensions.validationErrors[].code
+	// instead of a generic 500 from a handler-level fmt.Errorf.
```
Comment (tothandras) [inline api/v3/handlers/plans/convert.go]:
```suggestion
```

Comment (tothandras) [reply api/v3/handlers/plans/convert.go]: Not very useful.

**Review by tothandras — CHANGES_REQUESTED** [review]

**Review by tothandras — DISMISSED** [review]

**Review by tothandras — APPROVED** [review]
